koku/api/report/queries.py

```diff
@@ -239,49 +318,103 @@ def _get_time_frame_filter(self):
                 # get last 30 days
                 end = this_hour
                 start = ReportQueryHandler.n_days_ago(this_hour, 30)
+        self.start_datetime = start
+        self.end_datetime = end
+        self._create_time_interval()
+        return (self.start_datetime, self.end_datetime, self.time_interval)
+
+    def _get_filter(self):
+        """Create dictionary for filter parameters.
+
+        Returns:
+            (Dict): query filter dictionary
+
+        """
+        filter_dict = {
+            'usage_start__gte': self.start_datetime,
+            'usage_end__lte': self.end_datetime,
+        }
+        service = self.get_group_by_data('service')
```

Comment:
I was only targeting the group by service user story. I planned on adding the other group by mechanisms with follow on PRs, but was keeping it simple/focused for now instead of adding all of them as the test data will have to be come a bit more complicated.